### PR TITLE
doc: Clarify that feerates are per virtual size

### PR DESCRIPTION
**Pull request by maflcko**
By implementing segwit, it is already clear that all feerates in Bitcoin Core are denoted in (amount/virtual size). Though, there is  inconsistency, as some places use kvB, some use kB. Thus, replace all with "kvB".

See also commit 6da3afbaee5809ebf6d88efaa3958c505c2d71c7, which did the replacement for wallet RPCs.

### CODE REVIEW

**Review by jarolrod — COMMENTED** [review]
Concept ACK

What about: https://github.com/bitcoin/bitcoin/blob/2b45cf0bcdb3d2c1de46899e30885c953b57b475/src/qt/forms/sendcoinsdialog.ui#L853

``` diff
--- a/src/qt/forms/sendcoinsdialog.ui
+++ b/src/qt/forms/sendcoinsdialog.ui
@@ -850,7 +850,7 @@
                    <property name="toolTip">
                     <string>Specify a custom fee per kB (1,000 bytes) of the transaction's virtual size.
 
-Note:  Since the fee is calculated on a per-byte basis, a fee of "100 satoshis per kB" for a transaction size of 500 bytes (half of 1 kB) would ultimately yield a fee of only 50 satoshis.</string>
+Note:  Since the fee is calculated on a per-byte basis, a fee of "100 satoshis per kvB" for a transaction size of 500 virtual bytes (half of 1 kvB) would ultimately yield a fee of only 50 satoshis.</string>
                    </property>
                    <property name="text">
                     <string>per kilobyte</string>

```

**Comment by jonatack** [comment]
Concept ACK. I was hesitant to propose this across the codebase, as some reviewers prefered vB to kvB when I was working on the initial sat/vB `fee_rate` changes. A further possibility would be to replace "fee" with "fee rate" where the meaning is a fee rate and not a fee (credit to @Xekyo for tuning my antennae to this).

**Comment by jonatack** [comment]
There might be some remaining user-facing ones.
```cpp
src/init.cpp:541:    argsman.AddArg("-printpriority", strprintf("Log transaction fee per kB when mining blocks (default: %u)", DEFAULT_PRINTPRIORITY), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
src/wallet/rpcwallet.cpp:2295:                "\nSet the transaction fee per kB for this wallet. Overrides the global -paytxfee command line parameter.\n"
src/qt/forms/sendcoinsdialog.ui:851:                    <string>Specify a custom fee per kB (1,000 bytes) of the transaction's virtual size.
src/qt/forms/sendcoinsdialog.ui:853:Note:  Since the fee is calculated on a per-byte basis, a fee of "100 satoshis per kB" for a transaction size of 500 bytes (half of 1 kB) would ultimately yield a fee of only 50 satoshis.</string>
```
#20391 fixes the one in src/wallet/rpcwallet.cpp

**Comment by jarolrod** [comment]
@jonatack 
> src/qt/forms/sendcoinsdialog.ui:851:                    <string>Specify a custom fee per kB (1,000 bytes) of the transaction's virtual size.

That states `"of the transaction's virtual size"`

> src/qt/forms/sendcoinsdialog.ui:853:Note:  Since the fee is calculated on a per-byte basis, a fee of "100 satoshis per kB" for a transaction size of 500 bytes (half of 1 kB) would ultimately yield a fee of only 50 satoshis.</string>

Brought this one up here: https://github.com/bitcoin/bitcoin/pull/21752#pullrequestreview-649501091

**Comment by prayank23** [comment]
@jonatack 

> I was hesitant to propose this across the codebase, as some reviewers prefered vB to kvB when I was working on the initial sat/vB fee_rate changes. 

vB will be better IMO

> A further possibility would be to replace "fee" with "fee rate" where the meaning is a fee rate and not a fee

Agree

**Comment by MarcoFalke** [comment]
Thanks, done

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -601,8 +601,8 @@ static RPCHelpMan getnetworkinfo()
                                 {RPCResult::Type::BOOL, "proxy_randomize_credentials", "Whether randomized credentials are used"},
                             }},
                         }},
-                        {RPCResult::Type::NUM, "relayfee", "minimum relay fee for transactions in " + CURRENCY_UNIT + "/kB"},
-                        {RPCResult::Type::NUM, "incrementalfee", "minimum fee increment for mempool limiting or BIP 125 replacement in " + CURRENCY_UNIT + "/kB"},
+                        {RPCResult::Type::NUM, "relayfee", "minimum relay feerate for transactions in " + CURRENCY_UNIT + "/kvB"},
+                        {RPCResult::Type::NUM, "incrementalfee", "minimum feerate increment for mempool limiting or BIP 125 replacement in " + CURRENCY_UNIT + "/kvB"},
```
Comment (jonatack) [inline src/rpc/net.cpp:605]: ISTM the user-facing rpc and qt mostly use "fee rate"

Comment (MarcoFalke) [reply src/rpc/net.cpp]: Thanks, fixed

**Review by ryanofsky — APPROVED** [review]
Code review ACK fae196147bae11202c0d54543dc12ba5d92ab0cc. Checked instances where units were being added in the second commit and they all looked right.

File: src/qt/forms/sendcoinsdialog.ui [hunk 2]
```diff
@@ -850,7 +850,7 @@
                    <property name="toolTip">
                     <string>Specify a custom fee per kB (1,000 bytes) of the transaction's virtual size.
 
-Note:  Since the fee is calculated on a per-byte basis, a fee of "100 satoshis per kB" for a transaction size of 500 bytes (half of 1 kB) would ultimately yield a fee of only 50 satoshis.</string>
+Note:  Since the fee is calculated on a per-byte basis, a fee rate of "100 satoshis per kvB" for a transaction size of 500 virtual bytes (half of 1 kvB) would ultimately yield a fee of only 50 satoshis.</string>
```
Comment (rebroad) [inline src/qt/forms/sendcoinsdialog.ui:853]: Why the addition of the word "rate"? I thought the word "rate" usually meant over time, like a frequency. Not sure why we need the word here, given that a fee is a one to one relationship to a TX.

Comment (sipa) [reply src/qt/forms/sendcoinsdialog.ui:853]: We're everywhere referring to fee/(v)byte as feerate.

**Pull request merged**